src/fight/mod.rs

```diff
@@ -3,156 +3,301 @@
+    let defending_stats: &model::Stats = defending_regiment.get_model().get_stats();
+    let fastest: u8 = computation_tools::find_the_fastest(attacking_stats, defending_stats);
+    let first_damages: (usize, f64) = compute_case(attacking_regiment, defending_regiment, case);
+    let second_damages: (usize, f64) =
+        compute_case(defending_regiment, attacking_regiment, complementary);
+
+    match fastest {
+        0 => (
+            first_damages.0,
+            second_damages.0,
+            first_damages.1 * second_damages.1,
+        ),
+        1 => {
+            if first_damages.0 > defending_stats.get_health_point() {
+                (first_damages.0, 0, first_damages.1)
+            } else {
+                (
+                    first_damages.0,
+                    second_damages.0,
+                    first_damages.1 * second_damages.1,
+                )
+            }
+        }
+        2 => {
+            if second_damages.0 > attacking_stats.get_health_point() {
+                (0, second_damages.0, second_damages.1)
+            } else {
+                (
+                    first_damages.0,
+                    second_damages.0,
+                    first_damages.1 * second_damages.1,
+                )
+            }
+        }
+        _ => unreachable!("Code not supposed to be reached!"),
+    }
 }
 
-/// Return the fastest Model between 2 Model
-/// If the first Model is faster than the second Model, return 1
-/// If the second Model is faster than the first Model, return 2
-/// If the Models have the same speed, return 0
+/// ## Create a prediction from raw turn computation
 ///
-/// # Parameters
-/// attacking_model (&model::Model): The first Model
+/// ### Parameters
+/// (&regiment::Regiment) attacking_regiment -> The attacking regiment
 ///
-/// defending_model (&model::Model): The second Model
+/// (&regiment::Regiment) defending_regiment -> The defending regiment
 ///
-/// # Return
-/// usize: The fastest Model
-fn find_the_fastest(attacking_model: &model::Model, defending_model: &model::Model) -> usize {
-    let attacking_model_agility: usize = attacking_model.get_stats().get_agility() + 1;
-    let defending_model_agility: usize = defending_model.get_stats().get_agility();
-
-    match attacking_model_agility.cmp(&defending_model_agility) {
-        std::cmp::Ordering::Greater => 1,
-        std::cmp::Ordering::Less => 2,
-        std::cmp::Ordering::Equal => 0,
-    }
+/// (usize, usize, f64) raw_data -> The raw scenario computation
+///
+/// ### Return
+/// Prediction -> The scenario represented with the Prediction data structure
+fn turn_result_to_prediction(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+    raw_data: (usize, usize, f64),
+) -> prediction::Prediction {
+    let mut res_data: (Regiment, Regiment) =
+        (attacking_regiment.clone(), defending_regiment.clone());
```

Comment:
The cloning part is no more here but the point is that we are going to create 3 Predictions for 2 two units
So we need clones of our units before the fight so that we can report 3 different battle results